system/src/system_power_manager.cpp

```diff
@@ -505,6 +510,31 @@ void PowerManager::initDefault(bool dpdm) {
   clearIntermediateBatteryState(STATE_ALL);
 }
 
+void PowerManager::runDpdm() {
+// FIXME: maybe introduce a new flag?
+#if HAL_PLATFORM_POWER_WORKAROUND_USB_HOST_VIN_SOURCE
+  static system_tick_t lastRun = 0;
+  static uint8_t dpdmRetry = 0;
+
+  if (millis() - lastRun >= DPDM_PERIOD) {
+    lastRun = millis();
+    if (g_powerSource == POWER_SOURCE_VIN && HAL_USB_Get_State(nullptr) >= HAL_USB_STATE_POWERED) {
```

Comment:
what other states are caught by `HAL_USB_Get_State(nullptr) >= HAL_USB_STATE_POWERED`?  Can you make this more explicit such as a switch/case statement?